File: Práctica 1/S1R/pkg/S1R-0.0.1/lib/Carrera.rb
module Ruby
	public
	class Carrera
    
    attr_accessor:bicis,:codigo
    
		def clonar
      return Carrera.new(@codigo)
		end

		public
		def initialize(codigo)
      @codigo=codigo
			
		end
    
    public
    def aniadirparticipantes(bicicletas)
      @bicis=bicicletas
    end
    
    def mostrarranking
    end
    
	end
end

File: Práctica 1/S1/S1J/src/s1/Carrera.java

package s1;
import java.util.ArrayList;
import java.util.Collections;
    public abstract class Carrera extends Thread {
            protected ArrayList<Bicicleta> bicicletas;
            public int codigoCarrera;
            
            Carrera(int codigo){
                this.codigoCarrera = codigo;
                bicicletas = new ArrayList<>();
            }
            
            public ArrayList<Bicicleta> getBicicletas(){
                return this.bicicletas;
            }
            
            public void añadirParticipantes(ArrayList<Bicicleta> bicicletas){
                this.bicicletas = bicicletas;
            }
            
            public int getCodigo(){
                return codigoCarrera;
            }
            
            public void mostrarBicicletas(){
                for(int i=0;i<this.bicicletas.size();i++){
                    System.out.println(this.bicicletas.get(i).toString());
                }
            }
            
            public abstract void mostrarRanking();           
            
            public abstract void run();


    }